common/persistence/cassandra/mutable_state_task_store.go

```diff
@@ -809,6 +849,100 @@ func (d *MutableStateTaskStore) RangeCompleteVisibilityTask(
 	return gocql.ConvertError("RangeCompleteVisibilityTask", err)
 }
 
+func (d *MutableStateTaskStore) GetTieredStorageTask(
+	request *p.GetTieredStorageTaskRequest,
+) (*p.InternalGetTieredStorageTaskResponse, error) {
+	shardID := request.ShardID
+	taskID := request.TaskID
+	query := d.Session.Query(templateGetTieredStorageTaskQuery,
+		shardID,
+		rowTypeTieredStorageTask,
+		rowTypeTieredStorageTaskNamespaceID,
+		rowTypeTieredStorageTaskWorkflowID,
+		rowTypeTieredStorageTaskRunID,
+		defaultTieredStorageTimestamp,
+		taskID)
+
+	var data []byte
+	var encoding string
+	if err := query.Scan(&data, &encoding); err != nil {
+		return nil, gocql.ConvertError("GetTieredStorageTask", err)
+	}
+	return &p.InternalGetTieredStorageTaskResponse{Task: *p.NewDataBlob(data, encoding)}, nil
+}
+
+func (d *MutableStateTaskStore) GetTieredStorageTasks(
+	request *p.GetTieredStorageTasksRequest,
+) (*p.InternalGetTieredStorageTasksResponse, error) {
+
+	// Reading TieredStorage tasks need to be quorum level consistent, otherwise we could lose task
+	query := d.Session.Query(templateGetTieredStorageTasksQuery,
+		request.ShardID,
+		rowTypeTieredStorageTask,
+		rowTypeTieredStorageTaskNamespaceID,
+		rowTypeTieredStorageTaskWorkflowID,
+		rowTypeTieredStorageTaskRunID,
+		defaultTieredStorageTimestamp,
+		request.ReadLevel,
+		request.MaxReadLevel,
```

Comment:
maybe preferable to rename `ReadLevel` -> `MinTaskID`; `MaxReadLevel` -> `MaxTaskID`

Reply:
Done